--- /usr/src/linux/drivers/media/video/gspca/Kconfig.orig	2012-06-02 21:27:44.836124275 -0700
+++ /usr/src/linux/drivers/media/video/gspca/Kconfig	2012-06-02 21:28:28.678124370 -0700
@@ -199,6 +199,16 @@
 	 To compile this driver as a module, choose M here: the
 	 module will be called gspca_se401.
 
+config USB_GSPCA_EP800
+	tristate "EP800 USB Camera Driver"
+	depends on VIDEO_V4L2 && USB_GSPCA
+	help
+	 Say Y here if you want support for cameras based on the
+	 Endpoints (formerly known as AOX) ep800 chip.
+
+	 To compile this driver as a module, choose M here: the
+	 module will be called gspca_ep800.
+
 config USB_GSPCA_SN9C2028
 	tristate "SONIX Dual-Mode USB Camera Driver"
 	depends on VIDEO_V4L2 && USB_GSPCA
--- /usr/src/linux/drivers/media/video/gspca/Makefile.orig	2012-06-02 21:27:49.479124285 -0700
+++ /usr/src/linux/drivers/media/video/gspca/Makefile	2012-06-02 21:29:03.143124443 -0700
@@ -18,6 +18,7 @@
 obj-$(CONFIG_USB_GSPCA_PAC7302)  += gspca_pac7302.o
 obj-$(CONFIG_USB_GSPCA_PAC7311)  += gspca_pac7311.o
 obj-$(CONFIG_USB_GSPCA_SE401)    += gspca_se401.o
+obj-$(CONFIG_USB_GSPCA_EP800)    += gspca_ep800.o
 obj-$(CONFIG_USB_GSPCA_SN9C2028) += gspca_sn9c2028.o
 obj-$(CONFIG_USB_GSPCA_SN9C20X)  += gspca_sn9c20x.o
 obj-$(CONFIG_USB_GSPCA_SONIXB)   += gspca_sonixb.o
@@ -63,6 +64,7 @@
 gspca_pac7302-objs  := pac7302.o
 gspca_pac7311-objs  := pac7311.o
 gspca_se401-objs    := se401.o
+gspca_ep800-objs    := ep800.o
 gspca_sn9c2028-objs := sn9c2028.o
 gspca_sn9c20x-objs  := sn9c20x.o
 gspca_sonixb-objs   := sonixb.o
